src/main/java/com/example/provider/json/VehicleSerializer.java

```diff
@@ -28,36 +28,52 @@
 /**
  * Serializer for vehicle object to provide relevant information to its clients.
  *
- * <p>From the fleetengine vehicle object, this serializer retrieves the name and the vehicleState.
+ * <p>
+ * From the fleetengine vehicle object, this serializer retrieves the name and
+ * the vehicleState.
  */
 final class VehicleSerializer implements JsonSerializer<Vehicle> {
 
   @Override
   public JsonElement serialize(Vehicle src, Type typeOfSrc, JsonSerializationContext context) {
-    List<Waypoint> waypoints =
-        src.getWaypointsList().stream()
-            .map(
-                waypoint ->
-                    Waypoint.newBuilder()
-                        .setTripId(waypoint.getTripId())
-                        .setWaypointType(
-                            WaypointUtils.getWaypointTypeString(waypoint.getWaypointType()))
-                        .setLocation(
-                            SerializationUtils.createSerializedLocation(
-                                waypoint.getLocation().getPoint()))
-                        .build())
-            .collect(toList());
+    List<Waypoint> waypoints = src.getWaypointsList().stream()
+        .map(
+            waypoint -> Waypoint.newBuilder()
+                .setTripId(waypoint.getTripId())
+                .setWaypointType(
+                    WaypointUtils.getWaypointTypeString(waypoint.getWaypointType()))
+                .setLocation(
+                    SerializationUtils.createSerializedLocation(
+                        waypoint.getLocation().getPoint()))
+                .build())
+        .collect(toList());
+    SerializedLocation serializedLastLocation = null;
+    if (src.getLastLocation() != null) {
+      SerializedLatLng serializedLatLng = SerializedLatLng.newBuilder()
+      .setLatitude(src.getLastLocation().getLocation().getLatitude())
+      .setLongitude(src.getLastLocation().getLocation().getLongitude())
+      .build();
 
-    SerializedVehicle vehicle =
-        SerializedVehicle.newBuilder()
-            .setName(src.getName())
-            .setVehicleState(src.getVehicleState().name())
-            .setCurrentTripsIds(src.getCurrentTripsList())
-            .setWaypoints(waypoints)
-            .setBackToBackEnabled(src.getBackToBackEnabled())
-            .setSupportedTripTypes(src.getSupportedTripTypesList())
-            .setMaximumCapacity(src.getMaximumCapacity())
-            .build();
+      SerializedLocation.Builder serializedLastLocationBuilder = SerializedLocation.newBuilder()
+          .setPoint(serializedLatLng);
+
+      int vehicleHeading = src.getLastLocation().hasHeading() ? src.getLastLocation().getHeading().getValue() : 0;
```

Comment:
Similar comment on formatting, max line length is 100 https://google.github.io/styleguide/javaguide.html#s4.4-column-limit